### PR TITLE
Cache prover keys

### PR DESCRIPTION
**Pull request by mitschabaude**
* exposes `Pickles.Cache` along with types for the storables defined there
* adds an optional `storables` argument to `Pickles.compile` to override the existing storables
  * used here to customize for JS: https://github.com/o1-labs/o1js-bindings/pull/187
* implements `Pickles.Verification_key.of_yojson` to use as the serialization format

### CODE REVIEW

File: src/lib/pickles/cache.ml [hunk 1]
```diff
@@ -265,3 +276,19 @@ module Wrap = struct
     in
     (pk, vk)
 end
+
+module Storables = struct
```
Comment (mrmr1993) [inline src/lib/pickles/cache.ml:280]: This is unused in `Cache`; move it to `Pickles`

File: src/lib/pickles/cache.mli [hunk 2]
```diff
@@ -74,3 +106,14 @@ module Wrap : sig
          * [> `Cache_hit | `Generated_something | `Locally_generated ] )
          lazy_t
 end
+
+module Storables : sig
```
Comment (mrmr1993) [inline src/lib/pickles/cache.mli:110]: Ditto

File: src/lib/pickles/verification_key.ml [hunk 3]
```diff
@@ -100,6 +100,11 @@ module Verifier_index_json = struct
     verifier_index_to_yojson fp
       (fun _ -> `Null)
       (polycomm_to_yojson (or_infinity_to_yojson fq))
+
+  let of_yojson fp fq =
```
Comment (mrmr1993) [inline src/lib/pickles/verification_key.ml:104]: Nice :+1:

**Review by mrmr1993 — APPROVED** [review]
LGTM, but see nit

**Review by MartinMinkov — APPROVED** [review]

**Pull request merged**